category.go

```diff
@@ -100,10 +100,23 @@ func newFlagCategories() FlagCategories {
 
 func newFlagCategoriesFromFlags(fs []Flag) FlagCategories {
 	fc := newFlagCategories()
+
+	var categorized bool
 	for _, fl := range fs {
 		if cf, ok := fl.(CategorizableFlag); ok {
-			if cf.GetCategory() != "" {
-				fc.AddFlag(cf.GetCategory(), cf)
+			if cat := cf.GetCategory(); cat != "" {
```

Comment:
isnt this equivalent to ```fc.AddFlag(cf.GetCategory(), cf)``` with out the ```cf.getCategory() != "" ``` condition ?

Reply:
Yes, but then we're always working with categorized flags

Reply:
Ah ok got it.